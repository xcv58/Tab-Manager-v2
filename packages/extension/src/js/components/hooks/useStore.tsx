import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}

export const useFontSize = () => {
  const { userStore } = useStore()
  return userStore.fontSize
}

export const useTabHeight = () => {
  const fontSize = useFontSize()
  return 3 * fontSize
}
