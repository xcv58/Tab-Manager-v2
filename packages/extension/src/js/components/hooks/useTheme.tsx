import { createContext, useContext } from 'react'

export const ThemeContext = createContext(false)

export const useTheme = () => useContext(ThemeContext)
