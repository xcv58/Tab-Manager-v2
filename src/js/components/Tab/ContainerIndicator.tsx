import { observer } from 'mobx-react-lite'

let ContainerIndicator: any = () => null

if (process.env.TARGET_BROWSER === 'firefox') {
  ContainerIndicator = observer(require('./_ContainerIndicator').default)
}

export default ContainerIndicator
