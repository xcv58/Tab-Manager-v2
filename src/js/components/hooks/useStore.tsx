import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}
