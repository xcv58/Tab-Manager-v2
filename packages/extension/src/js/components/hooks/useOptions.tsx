import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return null
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  const { historyTabs } = searchStore
  console.log({ historyTabs: [...historyTabs] })
  return [...windowStore.tabs, ...historyTabs]
}
