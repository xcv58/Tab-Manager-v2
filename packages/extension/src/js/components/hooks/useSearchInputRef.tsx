import { useRef, useEffect } from 'react'
import { useStore } from 'components/hooks/useStore'

export const useSearchInputRef = () => {
  const searchInputRef = useRef<HTMLInputElement>(null)
  const { searchStore } = useStore()
  const { setSearchEl } = searchStore

  useEffect(() => setSearchEl(searchInputRef))
  return searchInputRef
}
