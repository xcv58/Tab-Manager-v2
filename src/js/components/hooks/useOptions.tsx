import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  return windowStore.tabs
}
