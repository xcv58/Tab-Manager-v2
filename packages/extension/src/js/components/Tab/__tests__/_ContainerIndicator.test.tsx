import React from 'react'
import * as StoreContext from 'components/hooks/useStore'
import ContainerIndicator from '../_ContainerIndicator'
import { render } from '@testing-library/react'

const VALID_COOKIE_STORE_ID = '42'
const props = {
  cookieStoreId: VALID_COOKIE_STORE_ID,
}

const mockStore = {
  containerStore: {
    getContainer: (cookieStoreId) => {
      if (cookieStoreId === VALID_COOKIE_STORE_ID) {
        return { colorCode: '#ff0000' }
      }
    },
  },
}

describe('ContainerIndicator', () => {
  beforeEach(() => {
    jest.spyOn(StoreContext, 'useStore').mockImplementation(() => mockStore)
  })

  it('render valid content if cookieStoreId is valid', () => {
    const { container } = render(<ContainerIndicator {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('render null for invalid cookieStoreId', () => {
    const { container } = render(<ContainerIndicator cookieStoreId="" />)
    expect(container).toMatchSnapshot()
  })
})
