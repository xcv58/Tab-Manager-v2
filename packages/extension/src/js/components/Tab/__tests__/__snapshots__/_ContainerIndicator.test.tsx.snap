// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerIndicator render null for invalid cookieStoreId 1`] = `<div />`;

exports[`ContainerIndicator render valid content if cookieStoreId is valid 1`] = `
<div>
  <hr
    class="absolute border"
    style="width: calc(100% - 24px); left: 12px; bottom: 2px; border-color: #ff0000;"
  />
</div>
`;
